### PR TITLE
[RDY] Replaces vacuum sealer requirements on sealed glass jar recipes with canning pot, to simulate water bath method

### PR DESCRIPTION
**Pull request by Regilith**
#### Summary

```SUMMARY: Balance "Removes the vac_sealer requirement from sealed glass jar recipes, adds canning pot, updates recipes."```

#### Purpose of change

Removes unrealistic vacuum sealer requirement for canned jar recipes.

~~The requirement was added with the introduction of sealed soups and such in #11040 without justification for it. You *can* vacuum seal jars of food, but it is unnecessary for boiled liquids because the act of boiling and canning automatically makes it self seal while also sterilizing it. It isn't necessary.~~

~~Source: https://nchfp.uga.edu/publications/publications_usda.html~~

~~Or just google how to can. You pour the stuff in the jar, put water in the pot, put the jar in the pot and boil for 15 min, which seals anything which has any liquid in it.~~

#### Describe the solution

Remove the vacuum sealer requirements from the recipes. Adds canning pot, makes recipes use canning pot instead, modify recipes to emulate canning via water bath method.

### CODE REVIEW

**Review by kevingranade — CHANGES_REQUESTED** [review]
See #22526 for some discussion of this.

Tl;dr
> There is no requirement in these recipes to have a sufficiently large pot to do waterbath canning canning, e.g. http://www.freshpreserving.com/waterbath-canning.html

If you want to make it that kind of recipe, you need to adjust the recipe to match how that kind of recipe works, not just make arbitrary adjustments. i.e. it needs a pot that is at least 2x the volume of the jar to be sealed, and require water to be used as part of the recipe. This complexity is part of the reason that this much simpler recipe was used.

Your source supports this, it calls for (optional) pre-cooking and (mandatory) water-bath sterilization.

**Comment by kevingranade** [comment]
> Do you want me to implement the traditional, open kettle style for .5L jars, and just have it require a pot, or should I add the canning pot?

Add the canning pot.  
That's going to be simpler anyway, open-kettle also requires tools to sterilize the jars.

**Comment by nexusmrsep** [comment]
Bucket (steel version) can be a fine option for this feature.

**Comment by hameleona** [comment]
@kevingranade You do not need to sterilize jars to seal them. Otherwise we should do it before we pour anything in them. It's not a sealing effort, it's a hygienic one.

**Comment by kevingranade** [comment]
> You do not need to sterilize jars to seal them.

It depends on the process, with water-bath canning, the jar is sterilized along with the food in the water bath.  With open kettle packing, the jar must be sterilized before use.  Since we dont have a way to track sterilization state of a jar, that

**Comment by kevingranade** [comment]
Means sterilization needs to happen in the same recipe.

**Comment by kevingranade** [comment]
Another option is adding a "canning" action attached to the new canning pot that lets you have custom logic for canning instead of it being shoehorned into the crafting interface.
It can actually compare the volume of the pot to the volume of the selected jars to determine how much water is needed, and try to do something intelligent with the water after it's done.

**Comment by Regilith** [comment]
I decided to go with Kevin's suggestion and just have it consume a flat amount of water, and abstract the water usage to keep the requirements sane in batch crafting. I modified the description of the pot to give a justification for it, where the survivor would fill the pot up with rocks or something to displace the water above the lid(s).

**Comment by Regilith** [comment]
Batch crafting seems to be good. It takes 2.5 hours to make 20 canned soup of all the recipes save bone soup, which takes longer. This pot would hold 6-7 jars at a time, You cook 20 soup in ~1 hour, and spend the next 1.5 canning it in 30 min intervals.

Sticking with the current description for the canning pot unless someone complains.

Gonna leave the cooking times unchanged, unless someone complains.

Gonna leave the cooking 3 requirement, as it kind of signifies that you are first cooking the soup in a pot, and then sealing it up with the canning pot.

**Comment by Regilith** [comment]
The timings might be in question, but this should be good to go.

**Comment by Mecares** [comment]
A regular bucket can contain 20 litres of water is there any reason it could not be used as container for those recipes? If not other items might be usable as well.

**Comment by Regilith** [comment]
A bucket contains 5L, not 20L.

Water bath canning requires a bit more specialized tools and setup than just submerging the jars in boiling water. Technically the jars need to be elevated ~1 inch above the bottom of the container to allow water to circulate beneath, so you would need to find some kind of rack to use with the bucket. The jars also need to be covered by at least 2 inches of water. You would normally also put a lid on top of the pot while it's cooking, which a bucket probably wouldn't have.

I could see it being handwaved, but then there would be no reason to get the canning pot, which is much more cumbersome, making the bucket strictly better.

The components are called using [ "steel_standard", 6 ], which is 6 steel lumps, 24 chunks of steel, or 120 scrap metal.

It would be 20 small metal sheets, or 6 lumps of steel. I'm ambivalent towards either.

**Comment by cake-pie** [comment]
I did the rationalization of sheet metal sizes #25210, so if I may offer some advice: depending on thickness, one sheet metal is roughly in the range of 70cm × 70cm × 1.53mm to 1m × 1m × 0.75mm ; small sheet metal is roughly 15 × 15cm to 20 × 20cm dimensions. This offers some room for variation / handwavium.

For your canning pot, a 25L vessel, you're looking at something like a [cylinder with diameter 30cm, height 36cm](https://www.wolframalpha.com/input/?i=cylinder+diameter+0.3m+height+0.36m) which has surface area approx 0.41m² (0.48m² with lid).

The most appropriate input for constructing it is probably a single sheet metal, cut and welded -- instead of solid pieces, which is what `steel_standard` represents.

Deconstruction should not return an intact sheet, so consider yielding a combination of small sheets for the pieces that you managed to cut out nicely, maybe a couple chunks of steel for the parts that ended up crumpled, and some scrap to cover the balance.

HTH.

**Comment by Regilith** [comment]
I was basing my numbers off the steel keg recipe, which takes 

> 13x lump of steel OR 52x chunk of steel OR 260x scrap metal 

to fabricate, and is twice the capacity.

I'm fine with replacing it with your suggestion, though finding an intact steel sheet would be a lot harder than just smashing a car for the necessary metal. It would probably be best to include both recipes.

I'm going to add a crucible component to the steel lump recipe, and add a sheet metal recipe in recipe_companions, and update the deconstruction to your suggestion.

**Comment by Regilith** [comment]
Removed the forging requirements, as what is essentially happening is the player is cutting the sheet metal into squares and welding them together into a box. It doesn't have to be pretty, it just has to function.

**Comment by cake-pie** [comment]
> the steel keg recipe

Yeah, #25210 only dealt with rationalizing recipes that were already using sheet metal, and did not include other recipes that were using solid steel in lieu of sheets. I really ought to get around to checking and updating those.

> finding an intact steel sheet would be a lot harder than just smashing a car for the necessary metal

Not really; instead of smashing (a lossy process that yields lumps, chunks, scrap) you can remove parts via the vehicle construction screen and thus reclaim the item they were constructed from. e.g. sawing off a quarterpanel will net you an intact steel sheet.

> Removed the forging requirements

You forgot the anvil and tongs.

**Comment by Regilith** [comment]
Updated

**Review by kevingranade — APPROVED** [review]
